src/code/audio_thread_manager.c

```diff
@@ -95,11 +154,12 @@ void AudioMgr_Init(AudioMgr* audioMgr, void* stack, OSPri pri, OSId id, Schedule
     audioMgr->irqMgr = irqMgr;
     audioMgr->rspTask = NULL;
 
-    osCreateMesgQueue(&audioMgr->taskQueue, audioMgr->taskMsgBuf, ARRAY_COUNT(audioMgr->taskMsgBuf));
+    osCreateMesgQueue(&audioMgr->taskDoneQueue, &audioMgr->taskDoneMsg, 1);
     osCreateMesgQueue(&audioMgr->interruptQueue, audioMgr->interruptMsgBuf, ARRAY_COUNT(audioMgr->interruptMsgBuf));
-    osCreateMesgQueue(&audioMgr->lockQueue, audioMgr->lockMsgBuf, ARRAY_COUNT(audioMgr->lockMsgBuf));
+    osCreateMesgQueue(&audioMgr->initQueue, &audioMgr->initMsg, 1);
 
-    osSendMesg(&audioMgr->taskQueue, NULL, OS_MESG_BLOCK);
+    // Send a message to the task queue so it is initially full
```

Comment:
```suggestion
    // Send a message to the task done queue so it is initially full
```
?

Also I know it's the same because these queues are only holding 1 message at most, but I find using "full" confusing instead of saying like "has 1 message available", but idk may just be me

Reply:
I don't find "has a message available" any more clear than just using "full" here. I'm open to describing this differently but I can't think of a satisfying way to word it currently.